projects/kit/components/slider/helpers/slider-key-steps.directive.ts

```diff
@@ -1,24 +1,59 @@
-import {Directive, forwardRef, inject, Input} from '@angular/core';
+import {
+    Directive,
+    forwardRef,
+    // eslint-disable-next-line no-restricted-syntax
+    HostBinding,
+    inject,
+    Input,
+    signal,
+} from '@angular/core';
+import type {TuiValueTransformer} from '@taiga-ui/cdk/classes';
 import {TuiControl} from '@taiga-ui/cdk/classes';
 import {tuiFallbackValueProvider} from '@taiga-ui/cdk/tokens';
-import {tuiClamp} from '@taiga-ui/cdk/utils/math';
 
 import {TuiSliderComponent} from '../slider.component';
 import type {TuiKeySteps} from './key-steps';
-import {tuiKeyStepValueToPercentage, tuiPercentageToKeyStepValue} from './key-steps';
+import {tuiCreateKeyStepsTransformer} from './key-steps';
 
 @Directive({
     standalone: true,
     selector: 'input[tuiSlider][keySteps]',
+})
+export class TuiSliderKeyStepsBase {
+    private readonly slider = inject<TuiSliderComponent>(
+        forwardRef(() => TuiSliderComponent),
+    );
+
+    // eslint-disable-next-line no-restricted-syntax
+    @HostBinding('attr.aria-valuemin')
+    protected min: number | null = this.slider.min();
+
+    // eslint-disable-next-line no-restricted-syntax
+    @HostBinding('attr.aria-valuemax')
+    protected max: number | null = this.slider.max();
+
+    public transformer = signal<TuiValueTransformer<number, number> | null>(null);
+
+    @Input()
+    public set keySteps(steps: TuiKeySteps | null) {
+        this.transformer.set(steps && tuiCreateKeyStepsTransformer(steps, this.slider));
+        this.min = steps?.[0][1] ?? null;
+        this.max = steps?.[steps.length - 1]?.[1] ?? null;
+    }
+}
+
+@Directive({
+    standalone: true,
+    selector:
+        'input[tuiSlider][keySteps][ngModel],input[tuiSlider][keySteps][formControl],input[tuiSlider][keySteps][formControl]',
     providers: [tuiFallbackValueProvider(0)],
     host: {
-        '[attr.aria-valuenow]': 'value()',
-        '[attr.aria-valuemin]': 'min',
-        '[attr.aria-valuemax]': 'max',
+        '[value]': 'this.value()',
+        '[attr.aria-valuenow]': 'transformer.toControlValue(this.value())',
         '[disabled]': 'disabled()',
         '(blur)': 'onTouched()',
-        '(input)': 'updateControlValue()',
-        '(change)': 'updateControlValue()',
+        '(input)': 'onChange($event.target.value)',
+        '(change)': 'onChange($event.target.value)',
     },
 })
 export class TuiSliderKeySteps extends TuiControl<number> {
```

Comment:
We should rename this to `TuiSliderKeyStepsControlDirective` and add an array `TuiSliderKeySteps = [base, control]`, otherwise people will not be attached without an explicit import

Reply:
Both directives are already part of 

```ts
export const TuiInputSlider = [...] as const;
```

Why do we need to create one more entity ?

Reply:
Oh, I forgot we already have this, sorry:
```ts
export const TuiSlider = [
    TuiSliderComponent,
    TuiSliderThumbLabel,
    TuiSliderKeySteps,
    TuiSliderReadonly,
] as const;
```